### DISCUSSION TITLE
Image route should not call function again

### DISCUSSION THREAD

**Discussion by Complexlity** [wevm/frog-archived#69] [Ideas]
Take this scenario

```ts
app.frame('/like', (c) => {
type Like = null | 'up' | 'down'
const {buttonValue: likeAction} = c

//old user like
let userLike: Like  = getUserLike(c) 

//new user like after preforming the action
userLike: Like =  await changeLike(oldUserLike, likeAction) 
// if like was null, insert new like, old and new are the same, remove like, if they are different, change like
return (
image: !userLike ? <EmptyLikeImage/> : <UpOrDownImage like={userLike}/>
intents: [
<Button value='up'>Upvote<Button/>,
<Button value='down'>DownVote<Button/>,
]
})
```

This is a simple component that should change the like of the user depending on what was clicked. 
If the user had no like and clicks the `up` button, here's what happens

1. POST request to the `/like` route and correctly upvotes
2. GET request to the `/like/` (`/like/image`) route but this time, the user would have an upvote in the db and the like action is also `up` so it remove the like
3. Returns `<EmptyLikeImage/>`  because you just remove the upvote while fetching the image

I found a work around

// don't run this function if the path is for the 'image'
if(path != "/like/image"){
userLike: Like =  await changeLike(oldUserLike, likeAction)
}

This finally works but it took a lot of frustration to figure out why my image was returning the wrong thing.

Suggestion: it would be better to not even call the entire function when the get request comes. So here's what the request would look like

1. POST request to the `/like` route and correctly upvotes
// returns frame data with fc:image: `/like/image?userLike=userLike`

2. GET request to the `/like` route but this time, called `/like/image?userLike=userLike` where the userLike would have been determined from the initial post route so the function technically doesn't run and just returns an image based on the searchParams

**Comment by jxom** [comment]
You can utilize `c.cycle` to distinguish: https://frog.fm/faq#why-is-my-frame-handler-being-invoked-twice
